resqpy/surface.py

```diff
@@ -1534,6 +1535,14 @@ def convert_to_irap(self, file_name):
          for item in lines:
             f.write(item)
 
+   def append_extra_metadata(self, meta_dict):
+      """Append a given dictionary of metadata to the existing metadata."""
+
+      if self.extra_metadata is None:
+         self.extra_metadata = {}
+      for key in meta_dict:
+         self.extra_metadata[key] = meta_dict[key]
```

Comment:
1. Would be great to have a unit test
2. Is it possible for extra_metadata to be None? I think it might be an always present dictionary
3. To merge dicts in python 3.5+ one can use `z = {**x, **y}`. But either way is fine, explicit is good too I think!